src/WidgetBase.ts

```diff
@@ -291,6 +293,33 @@ export class WidgetBase<P = WidgetProperties, C extends DNode = DNode> extends E
 		return this._properties;
 	}
 
+	@diffProperty('registry', reference)
+	protected diffPropertyRegistry(previousProperties: any, newProperties: any): void {
+		if (this._registries.defaultRegistry === this._defaultRegistry) {
+			this._registries.remove(this._defaultRegistry);
+		}
+		const result = this._registries.replace(previousProperties.registry, newProperties.registry);
+		if (!result) {
+			this.getRegistries().add(newProperties.registry);
```

Comment:
Why access the public method here, when in the rest of the code, you are not accessing that way, but instead directly to the private member?